tests/test_example_repo.py

```diff
@@ -123,3 +123,40 @@ def test_no_git_no_warning(capsys):
         os.environ['PATH'] = path_backup
     for k, v in data.items():
         assert v == ['None', 'undefined']
+
+
+def test_exclude_patterns_srcdir_relative():
+    data = run_sphinx(
+        'repo_full',
+        git_exclude_patterns='I 🖤 Unicode.rst',
+    )
+    assert data == {
+        **expected_results,
+        'I 🖤 Unicode': ['None', 'undefined'],
+    }
+
+
+def test_exclude_patterns_glob():
+    data = run_sphinx(
+        'repo_full',
+        git_exclude_patterns='**/*.rst',
```

Comment:
This could be shortened to ...

```suggestion
        git_exclude_patterns='**.rst',
```

... but I'm not sure if that's better.
But it would behave exactly the same, right?

The test repo has no subdirectories, so even `'*.rst'` should be sufficient, but `'**.rst'` could be considered more "future-proof" if directories are added later.

It would have been good to have subdirectories in the test repo, but I don't think it's worth adding them just for that.

Reply:
> This could be shortened to ...
>
>     git_exclude_patterns='**.rst',
>
> ... but I'm not sure if that's better. But it would behave exactly the same, right?

Based on <https://www.sphinx-doc.org/en/master/usage/configuration.html#id12>:

>  A double star ** can be used to match any sequence of characters including slashes.

And <https://github.com/sphinx-doc/sphinx/blob/5e9550c78e3421dd7dcab037021d996841178f67/sphinx/util/matching.py#L60-L65>:

```python
class Matcher:
    """A pattern matcher for Multiple shell-style glob patterns.
    Note: this modifies the patterns to work with copy_asset().
          For example, "**/index.rst" matches with "index.rst"
    """
```

Yes it would behave exactly the same.
I originally used this pattern to make sure that the `**` was working correctly, but as there is no subdirectory this is not a very effective test.

> The test repo has no subdirectories, so even `'*.rst'` should be sufficient, but `'**.rst'` could be considered more "future-proof" if directories are added later.

Maybe this is a good argument to choose `'*.rst'` instead as this way the validity of the test is less likely to change overtime.

> It would have been good to have subdirectories in the test repo, but I don't think it's worth adding them just for that.

I don't think so either. The Matcher from Sphinx is probably sufficiently tested on their side.